core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -883,8 +903,8 @@ private object RTS {
       })
 
     @inline
-    final def shouldDie: Option[Throwable] =
-      if (!killed || noInterrupt > 0) None else status.get.error
+    final def shouldDie: Option[List[Throwable]] =
```

Comment:
Since you are not calling `get` on this, may as well just return a boolean from `shouldDie`.